packages/zapp/console/src/components/flytegraph/ReactFlow/transformDAGToReactFlowV2.tsx

```diff
@@ -78,8 +80,16 @@ const buildReactFlowDataProps = (props: BuildDataProps) => {
   const nodeExecutionStatus = mapNodeExecutionStatus();
 
   // nodeExecutionsById null check is required as on first render it can be undefined
+  const mapNodeExternalResources = () => {
+    if (nodeExecutionsById && nodeExecutionsById[node.scopedId]) {
+      return nodeExecutionsById[node.scopedId]?.externalResourcesByPhase;
```

Comment:
nit: you don't need to check for null on line 85, the if condition above ensures it has a value